Feature/additional unit tests

alexanderbez opened the pull request:
### Problem

No unit tests for the `EllipticCurve` or `EllipticCurvePoint` types to test basic functionality.

### Solution

Implement base unit tests for the aforementioned types in the `utility` package.

### Remember

- Bump `VERSION`
- Update `CHANGELOG.md`
- Update `README.md`
- Write tests

Commit:
Add "0x" prefix to expected hash results in test client fixtures

Commit:
Add style formatting consistency to client tests

Commit:
Move local scope client test fixtures to fixtures file

Commit:
Version bump 1.6.1

Commit:
Update CHANGELOG

Commit:
Merge branch 'master' of github.com:CityOfZion/neo-go-sdk into feature/additional-unit-tests

Commit:
Implement unit tests for NewModularArithmetic

Commit:
Add TODO in ModularArithmetic#Sqrt to handle error(s)

Commit:
Implement unit tests for EllipticCurvePoint

Commit:
Rename ModularArithmetic unit test function names

Commit:
Implement unit tests for EllipticCurve

Commit:
Merge branch 'master' of github.com:CityOfZion/neo-go-sdk into feature/additional-unit-tests

Commit:
Version bump 1.8.1

Commit:
Update CHANGELOG

revett commented:
@alexanderbez Like the look of this so far! Let me know when it's ready for a review 👍

Commit:
Fix testTransactionOutputs test fixture invalid hash and value

Commit:
Update CHANGELOG

Commit:
Implement utility test functions for the utility package

Commit:
Update ModularArithmetic unit tests to actually use big integers

Commit:
Update CHANGELOG

alexanderbez commented:
@revett I updated a broken client test fixture (invalid hash and value). This was causing the build to fail. I also updated the modular arithmetic tests to use actual big ints.

Everything else seems to gave solid unit/integrations tests in place. If there are any that you think are missing, let me know and I'd be happy to write em' up. Otherwise, I think this PR is g2g for review 👍

revett commented:
@alexanderbez Great - will review ASAP 👌

alexanderbez commented:
Actually, @revett let me update the mod arithmetic unit tests some more. Some of them aren't using truly big ints. Will update once I finish 👍

revett commented on utility/modular_arithmetic.go:
```diff
@@ -48,6 +48,9 @@ func (m ModularArithmetic) Mul(x *big.Int, y *big.Int, p *big.Int) (z *big.Int)
 
 // Sqrt computes: sqrt(x) % p.
+//
+// TODO: I believe we should return an error here and bubble that upstream.
```
Can you raise an issue for this and then remove the comment? We can address it in a later PR.

alexanderbez replied:
Sure thing!

revett commented on utility/utility_test.go:
```diff
@@ -0,0 +1,12 @@
+// hexToInt accepts a hex value represented as a string and returns an
+// initialized big integer with the provided hex value ignoring any errors.
+func hexToInt(h string) (i *big.Int) {
+	i, _ = new(big.Int).SetString(h, 16)
```
Why ignore errors?

alexanderbez replied:
This function would only be used by unit tests and those unit tests should only pass valid hex strings 👍

revett reviewed (changes requested):
Few questions/comments 👍

Commit:
Remove TODO for error handling in ModularArithmetic#Sqrt
